spec/services/update_team_name_and_ldu_service_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+RSpec.describe UpdateTeamNameAndLduService do
+  context "when the team has the wrong name" do
```

Comment:
I think you can reduce the duplication by using `let` here.

I couldn't resist playing with this, here's where I got to. You definitely don't have to use this verbatim, just to explain my thoughts!

```ruby
describe UpdateTeamNameAndLduService do
  context "with an existing LDU and team" do
    before do
      create(:local_divisional_unit, code: 'LDU1')
      team = build(:team, code: team_code)
      team.local_divisional_unit = local_divisional_unit
      team.save(validate: false)
    end

    context "when the team is not associated with the LDU" do
      let(:team_code) { 'TEAM1'}
      let(:local_divisional_unit) { nil }

      it "repairs the association" do
        described_class.update(team_code: team_code, ldu_code: 'LDU1')
        team = Team.find_by(code: team_code)
        expect(team.local_divisional_unit.code).to eq('LDU1')
      end
    end

    context "when the team is associated with a different LDU" do
      let(:team_code) { 'TEAM1'}
      let(:local_divisional_unit) { create(:local_divisional_unit, code: 'other') }

      it "associates the team with the new LDU" do
        described_class.update(team_code: team_code, ldu_code: 'LDU1')
        team = Team.find_by(code: 'TEAM1')
        expect(team.local_divisional_unit.code).to eq('LDU1')
      end
    end

    context "when the team is already associated to the correct LDU" do
      let(:team_code) { 'TEAM1'}
      let(:local_divisional_unit) { create(:local_divisional_unit) }

      it "doesn't update the team record" do
        described_class.update(team_code: team_code, ldu_code: local_divisional_unit.code)
        expect_any_instance_of(Team).not_to receive(:save)
        described_class.update(team_code: 'TEAM1', ldu_code: 'LDU1')
      end
    end
  end
end
```

Reply:
(note that this doesn't cover all of the tests)